includes/admin/class-amp-story-templates.php

```diff
@@ -65,20 +65,28 @@ public function filter_user_has_cap( $allcaps, $caps, $args ) {
 	}
 
 	/**
-	 * Temporarily filter pre_get_posts to not display templates in the list of reusable blocks.
+	 * Ensures that the templates only display in the AMP Story editor.
+	 *
+	 * This filters pre_get_posts to not display templates in the list of reusable blocks,
+	 * or anywhere other than in the AMP Story editor.
 	 *
 	 * @param object $query WP_Query object.
 	 * @return object WP Query modified object.
 	 */
 	public function filter_pre_get_posts( $query ) {
-		global $pagenow;
-
-		if ( 'edit.php' !== $pagenow || ! $query->is_admin ) {
+		if ( 'wp_block' !== $query->get( 'post_type' ) ) {
 			return $query;
 		}
 
-		if ( 'wp_block' !== $query->get( 'post_type' ) ) {
-			return $query;
+		$referer = wp_parse_url( wp_get_referer() );
+		if ( isset( $referer['query'] ) ) {
+			if ( 'post_type=' . AMP_Story_Post_Type::POST_TYPE_SLUG === $referer['query'] ) {
```

Comment:
I'd use `$parsed_args = wp_parse_args( $referer['query'] );` already here, and then compare with `$parsed_args['post_type']` directly. Otherwise this check fails when some middleware adds additional query params to the request.